provider/internal/queue/queue.go

```diff
@@ -15,144 +17,141 @@ import (
 
 var log = logging.Logger("provider.queue")
 
-// Queue provides a best-effort durability, FIFO interface to the datastore for storing cids
+const (
+	// Number of input CIDs the queue will hold..
+	defaultQueueCapacity = 65536
+	// Time for Close to wait to finish writing CIDs to datastore.
+	shutdownTimeout = 5 * time.Second
```

Comment:
```suggestion
	shutdownTimeout = 20 * time.Second
```

? Systemd waits ups to 1m30s when something doesn't want to stop, so I'm guessing we are still way more polite to the user.

Reply:
Done